### DISCUSSION TITLE
`.extensions()` method not found in `HttpRequest`

### DISCUSSION THREAD

**Discussion by mohd109** [actix/actix-web#3204] [General] [answered]
Hi everyone,
first of all thanks for such a good library
I have a problem regarding the extensions() method which is not found in the HttpRequest

## Expected Behavior

Request.extensions should work just like in the examples

## Current Behavior

but it's not found in the httprequest

```
Identity::login(&request.extensions(), user._id.to_owned().to_string()).unwrap();
                         ^^^^^^^^^^ method not found in HttpRequest
```

## Steps to Reproduce (for bugs)

I just updated to the last version and used actix-identity example for redis storage. Everything is ok. Just the extensions method is not found in the compile time. 

## Your Environment
both in Ubuntu 22 and windows server 2022
vscode

- Rust Version (1.74`):
- Actix Web Version:
- 
actix-web = { version = "4.4.0", features = ["openssl"] }
actix-session = { version = "0.8.0", features = ["redis-rs-session","redis-actor-session"] } 
actix-redis =  "0.13.0"
actix-identity = "0.6.0"

**Comment by robjtede** [comment]
`.extensions()` is part of the `HttpMessage` trait, you need to `use` it.

```diff
+ use actix_web::HttpMessage;
```

**Reply by mohd109** [reply]
Yeah I know, but in the examples it's from httprequest. Do we have any examples of httpmessage for identity::login?

**Reply by robjtede** [reply, marked as answer]
I assume you mean this line: https://github.com/actix/examples/blob/1b43dde946ea78a94aab98555a30f62bb506bdcf/auth/simple-auth-server/src/auth_handler.rs#L56

Yes, that is coming from the `HttpMessage` implementation on `HttpRequest`. You just need to add the `use` line I showed above. (See [it being `use`d on line 5 of that same file here](https://github.com/actix/examples/blob/1b43dde946ea78a94aab98555a30f62bb506bdcf/auth/simple-auth-server/src/auth_handler.rs#L5).)
